profiling/src/io.rs

```diff
@@ -491,14 +433,10 @@ unsafe extern "C" fn observed_fread(
     let len = ORIG_FREAD(ptr, size, nobj, stream);
     let duration = start.elapsed();
 
-    FILE_READ_TIME_PROFILING_STATS.with(|cell| {
-        let mut io = cell.borrow_mut();
-        io.track(duration.as_nanos() as u64)
-    });
-    FILE_READ_SIZE_PROFILING_STATS.with(|cell| {
-        let mut io = cell.borrow_mut();
-        io.track(len as u64)
-    });
+    FILE_READ_TIME_PROFILING_STATS.with_borrow_mut(|io| io.track(duration.as_nanos() as u64));
+    if len > 0 {
```

Comment:
Most other place guard this with a `len > 0` check, but this one didn't. Added for clarity, let me know if it's not desired.

Reply:
This seems to make sense :-)

Reply:
Your patch is correct, thanks